src/include/update/rootpart.rb

```diff
@@ -505,6 +506,15 @@ def RootPartitionDialog(flavor)
             end
           end
         end
+
+        if ret != :back
```

Comment:
well, method is already really big. I would like to see some improvements in that direction here.

Also identical condition is directly above, so why repeating it here?

Reply:
It's repeated because it can be set to `:back` inside in the error handler. We would need to test it again anyway.

I'm quite reluctant to do more refactoring as I already spent a lot of time elsewhere and this is a P1 blocking the QA, I'd like to have it out ASAP...

Reply:
ok